catalog/solr/catalog-solr-core/src/main/java/ddf/catalog/source/solr/SolrMetacardClientImpl.java

```diff
@@ -404,9 +447,9 @@ private SolrDocumentList handleSpellcheck(
       LOGGER.trace("End solr spellcheck: query request {}", request);
     }
     if (highlightResponse.getHighlighting() != null) {
-      LOGGER.trace("Starting highlight extraction: query request {}", request);
+      LOGGER.trace("Starting highlight extraction.");
       highlighter.processPostQuery(highlightResponse, responseProps);
-      LOGGER.trace("Ending highlight extraction: query request {}", request);
+      LOGGER.trace("Ending highlight extraction.");
```

Comment:
❓ There are a decent amount of assumptions that this processing could be taking a while.  We can obviously extract out the timing based on the trace log statements between these two log statements but there is no way to associate those numbers with a given query and this also would only be logged if the logging level is bumped up to trace.  Is there a way to incorporate this execution time into the collected metrics so that we could tell the time difference between the spellcheck handling (which can include a whole other query to solr) vs the highlight processing?

Reply:
Think I missed this one. Added timing for the `highlighter.processPreQuery` but missed this `processPostQuery`. Went through the code again and added timing around all significant methods.